src/MPoly.jl

```diff
@@ -16,6 +16,32 @@ coefficient_ring(a::MPolyRingElem) = base_ring(a)
 
 coefficient_ring(R::MPolyRing) = base_ring(R)
 
+@doc md"""
+    mpoly_type(::Type{T}) where T<:RingElement
+
+The type of multivariate polynomials with coefficients of type `T`.
+Falls back to `Generic.MPoly{T}`.
+
+# Examples
+```julia
+mpoly_type(typeof(ZZ()))
+```
+"""
+mpoly_type(::Type{T}) where T<:RingElement = Generic.MPoly{T}
```

Comment:
Meta question / suggestion: Should we have alternate methods for all these `_type` functions that take a regular object and then re-dispatch using its type? I.e.:
```julia
mpoly_type(x::T) where T<:RingElement = mpoly_type(T)
```
and so on. That might be convenient in some cases?

Or maybe it's a bad idea, dunno

Reply:
Like here
https://github.com/oscar-system/Oscar.jl/blob/2e81551de2ecfc38730eaebed97ec974e54a48b4/src/Rings/mpoly_types.jl#L17-L20
?

Reply:
Ah, indeed. Then it's better to do it here, once, for all types, isn't it?

Reply:
Yes, I agree. I am not sure we want to add the one with the parents, e.g. `mpoly_type(::Type{QQField}) = QQMPolyRingElem`.

Reply:
Done in PR #1290.

> I am not sure we want to add the one with the parents, e.g. `mpoly_type(::Type{QQField}) = QQMPolyRingElem`.

Asking for the multivariate polynomial type of `QQField` makes sense. So we could add
```julia
mpoly_type(::Type{T}) where T<:Ring = mpoly_type(elem_type(T))
dense_poly_type(::Type{T}) where T<:NCRing = dense_poly_type(elem_type(T))
```
This would let us simplify calls of `*poly_type(elem_type(T))` to `*poly_type(T)`. Anyway, there are only very few calls of those functions in AbstractAlgebra, Nemo, Hecke and Oscar at all.

Reply:
But then you also want it for an instance of the ring.